src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -20,36 +21,49 @@ public class DeleteCommand extends Command {
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Parameters: INDEX [OPTIONAL INDEXES] (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deletion successful!";
+    public static final String NO_PERSON_TO_DELETE_ERROR = "There is no student to delete.";
 
-    private final Index targetIndex;
+    private final List<Index> targetIndexes;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    /**
+     * Creates a DeleteCommand to delete the specified {@code Person}(s).
+     *
+     * @param targetIndexes of the person(s) to delete.
+     */
+    public DeleteCommand(List<Index> targetIndexes) {
+        Collections.sort(targetIndexes);
+        this.targetIndexes = targetIndexes;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        if (model.getFilteredPersonList().size() == 0) {
+            throw new CommandException(NO_PERSON_TO_DELETE_ERROR);
         }
 
-        Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
-        model.deletePerson(personToDelete);
-        ClassStorage.removeExistingClass(personToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, personToDelete));
+        for (int i = targetIndexes.size() - 1; i >= 0; i--) {
+            Index targetIndex = targetIndexes.get(i);
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+            Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            model.deletePerson(personToDelete);
+            ClassStorage.removeExistingClass(personToDelete);
+        }
+        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS));
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof DeleteCommand // instanceof handles nulls
-                && targetIndex.equals(((DeleteCommand) other).targetIndex)); // state check
+                && targetIndexes.equals(((DeleteCommand) other).targetIndexes)); // state check
```

Comment:
The`equals` method of `List` returns `True` when the two `List` are of the same size and corresponding items are equal to each other. In this case, should we allow `delete 1 2` be equal to `delete 2 1`?

Reply:
I think this wouldn't affect the delete function since `delete 1 2` and `delete 2 1` work the same way (successfully deletes the first two students in the list). Does this make sense?

Reply:
The same logic applies here, since the `targetIndexes` must always be sorted, there would not be a difference then.